experiment/runner.py

```diff
@@ -177,23 +177,36 @@ def run_fuzzer(max_total_time, log_filename):
     runner_niceness = environment.get('RUNNER_NICENESS', 0)
 
     try:
-        with open(log_filename, 'w') as log_file:
-            # Because the runner is launched at a higher priority,
-            # set it back to the default(0) for fuzzing processes.
-            new_process.execute([
-                'nice', '-n',
-                str(0 - runner_niceness), 'python3', '-u', '-c',
-                ('import fuzzer; '
-                 'fuzzer.fuzz('
-                 "'{input_corpus}', '{output_corpus}', '{target_binary}')"
-                ).format(input_corpus=shlex.quote(input_corpus),
-                         output_corpus=shlex.quote(output_corpus),
-                         target_binary=shlex.quote(target_binary))
-            ],
+        # Because the runner is launched at a higher priority,
+        # set it back to the default(0) for fuzzing processes.
+        command = [
+            'nice', '-n',
+            str(0 - runner_niceness), 'python3', '-u', '-c',
+            ('import fuzzer; '
+             'fuzzer.fuzz('
+             "'{input_corpus}', '{output_corpus}', '{target_binary}')").format(
+                 input_corpus=shlex.quote(input_corpus),
+                 output_corpus=shlex.quote(output_corpus),
+                 target_binary=shlex.quote(target_binary))
+        ]
+
+        fuzzer_environment = _get_fuzzer_environment()
+        # Write output to stdout if user is fuzzing from command line.
+        # Otherwise, write output to the log file.
+        if environment.get('FUZZ_OUTSIDE_EXPERIMENT'):
+            new_process.execute(command,
                                 timeout=max_total_time,
-                                output_files=[log_file],
+                                output_file=sys.stdout.fileno(),
```

Comment:
i dont like this, i think output_file should either mean None (no output file) or a file handle.
right now, we are doing
default - PIPE
this one - stdout print
output_file - file handle

i think we should just have another argument
print_output and hide this stuff inside newprocess.execute itself
if print_output:
 output_file = sys.stdout.fileno()
elif not output_file:
  output_file = subprocess.PIPE
then use,

kwargs['stdout'] = output_file
kwargs['stderr'] = subprocess.STDOUT